mesonbuild/backend/ninjabackend.py

```diff
@@ -36,7 +36,7 @@
 from .backends import CleanTrees
 from ..build import InvalidArguments
 
-if mesonlib.is_windows():
+if mesonlib.is_cmd_shell():
```

Comment:
This looks like it should already be right for the case you are fixing...

`is_windows()` is supposed to return true if the build machine is windows, not if we are targetting windows (c.f. `for_windows()`

Reply:
Hmmm, oops, I must have just seen the shell escape issues and then also assumed I needed to update the check for what shell is in use. I think with my first experiments I deleted the first is_windows() path just to be sure I knew the backend code would assume an sh shell. When I inspected the implementation of `is_windows()` I was thinking that `platform.system()` was pulling out the 'system' from the cross file but that's just the regular python platform api, oops.  I just double checked that after reverting this patch I could still do my cross-compile ok, so I'll drop this patch from the PR - sorry for the noise.

Reply:
> sorry for the noise.

No problem.  This is why reviews are a good idea :)